src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx

```diff
@@ -230,36 +244,42 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
 
     return (
         <ForceGraph2D
-            graphData={data}
             ref={fgRef} // fix graph to canvas
+            graphData={data}
             nodeLabel={"nodeType"} // tooltip on hover, actual label is in nodeCanvasObject
+            backgroundColor={"#18191d"}
+            height={700}
+            width={1000}
+            onEngineStop={() => {
+                if (!stopEngine) {
+                    fgRef.current.zoomToFit(1000, 50);
+                    setStopEngine(true);
+                }
+            }}
             nodeCanvasObject={nodeStyling}
             nodeCanvasObjectMode={() => "after"}
             onNodeHover={nodeHover}
             onNodeClick={nodeClick}
+            onNodeDrag={(node) => {
+                node.fx = node.x;
+                node.fy = node.y;
+            }}
             onNodeDragEnd={(node) => {
                 node.fx = node.x;
                 node.fy = node.y;
             }}
             linkColor={(link) =>
                 highlightLinks.has(link)
                     ? colors.highlightLink
-                    : calcLinkColor(link as Link, data as VizGraph)
+                    : // : "#555"
+                      calcLinkColor(link as Link, data as VizGraph)
```

Comment:
does this imply that the above `link: any` line 197 should be a `link: Link` ? I could be wrong

Reply:
Will resolve in a separate PR - we need a few new types to distinguish types for the Graph and types that come back as raw data. These changes are more than just a few lines and don't exactly fit into the scope of this PR, but I will address this. Thanks!